golem_messages/message.py

```diff
@@ -140,13 +139,23 @@ def raw(self):
         """Returns a raw copy of the message"""
         return self._raw[:]
 
-    def get_short_hash(self):
+    def get_short_hash(self, payload=None):
         """Return short message representation for signature
         :return bytes: sha1(TYPE, timestamp, encrypted, payload)
         """
+        if payload is None:
+            payload = serializer.dumps(self.slots())

```

Comment:
This is a step back and introduces a constant overhead. It means that when we sign message's hash, we serialize the already serialized slots. We also serialize slots of a message when we want to verify the signature, when the message was deserialized a moment ago.

Reply:
It won't happen when signing (the `payment` arg is passed). It could be optimized in verification stage, but at this moment I think it's negligible. And removing `_payload` cache slot saves us from all cache related troubles.

Reply:
Golem was profiled and the overhead of re-serialization wasn't negligible. That's why this change was introduced.

Reply:
#55 Thanks for pointing this out.